### DISCUSSION TITLE
2fa does not work; KeepassXC has no TOTP functionalty; 6-digit code always wrong

### DISCUSSION THREAD

**Discussion by electric-socket** [community/community#118410] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am trying (and failing) to use 2FA to login and it does not work. I copied the secret token Github gave me, gave it to the TOTP desktop application I am using (Verifyr). The Program runs on the same computer I'm logging in from, so it cannot be a time discrepancy. I always grab a code that has 30 seconds before it expires and immediately type it in, and it always fails. I don't always have my cell phone with me, and can't get to it if it's not in immediate reach (i am a paraplegic and bedridden), so using a cell phone is not an option.  

I tried KeepassXC as this site recommended, but it is only a password manager, it does not have TOTP functionality. I don't need a  password manager, (Firefox does that just fine.) I need a  TOTP program! I just want to log in and access the site. Something is wrong and I have no idea what it is. 

It doesn't matter if I log on using the same browser that I requested the secret on, but that makes no difference. Beyond that, 2FA is supposed to be tied to the token generated, not to the specific browser. 

Why does this have to be so hard? What's wrong with using passphrase? I dislike that too, but it's less of a pain than being asked for a token that is then rejected.

So my question is: How do I get this to work?

In the mean time, I'll keep using recovery codes, and if that's the only solution forever, so be it. But 1, it's irritating and 2, I'm afraid they'll take that away from me too and I'll have no way to logon. "No good deed goes unpunished."

Paul
----------------------------
Paul Robinson <paul@paul-robinson.us> 
"The lessons of history teach us - if they teach us anything - is that no one learns the lessons that history teaches us."

**Comment by jfullstackdev** [comment]
answer from Copilot, might be of help

---

I'm sorry to hear about the trouble you're having with 2FA. KeePassXC does indeed support TOTP, but it's a bit hidden. Here's how you can set it up:

1. Open KeePassXC and go to the entry for GitHub.
2. Right-click on the entry and select "Edit Entry".
3. Go to the "Advanced" tab.
4. Click on "Add" under "Additional Attributes".
5. For the name, enter "TOTP Seed" (without the quotes).
6. For the value, enter the secret key that GitHub gave you.
7. Check the box for "Protect".
8. Click "OK" to close the dialog.
9. Now, you should see a "Show TOTP" button in the toolbar when you select the GitHub entry.

Every time you need a TOTP code, you can select the GitHub entry and click "Show TOTP" to generate a new code.

If you're still having trouble, it might be worth trying a different TOTP application to see if that resolves the issue. There are many available, such as Authy (which has a desktop application), WinAuth, or Aegis (if you ever have access to an Android device).

If none of these solutions work, you might want to contact GitHub support directly for assistance. They may be able to help troubleshoot the issue or provide an alternative solution.
